src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -67,43 +67,63 @@ function WorkspaceCategoriesPage({policy, policyCategories, route}: WorkspaceCat
-                                {value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}
-                            </Text>
-                            <View style={[styles.p1, styles.pl2]}>
-                                <Icon
-                                    src={Expensicons.ArrowRight}
-                                    fill={theme.icon}
-                                />
+                .map((value) => {
+                    const isDisabled = value.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || Object.values(value.pendingFields ?? {}).length > 0;
+                    return {
+                        text: value.name,
+                        keyForList: value.name,
+                        isSelected: !!selectedCategories[value.name],
+                        isDisabled,
+                        pendingAction: value.pendingAction,
+                        errors: value.errors ?? undefined,
+                        rightElement: (
+                            <View style={[styles.flexRow, isDisabled && styles.buttonOpacityDisabled]}>
+                                <Text style={[styles.textSupporting, styles.alignSelfCenter, styles.pl2, styles.label]}>
+                                    {value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}
+                                </Text>
+                                <View style={[styles.p1, styles.pl2]}>
+                                    <Icon
+                                        src={Expensicons.ArrowRight}
+                                        fill={theme.icon}
+                                    />
+                                </View>
                             </View>
-                        </View>
-                    ),
-                })),
-        [policyCategories, selectedCategories, styles.alignSelfCenter, styles.flexRow, styles.label, styles.p1, styles.pl2, styles.textSupporting, theme.icon, translate],
+                        ),
+                    };
+                }),
+        [
+            policyCategories,
+            selectedCategories,
+            styles.alignSelfCenter,
+            styles.buttonOpacityDisabled,
+            styles.flexRow,
+            styles.label,
+            styles.p1,
+            styles.pl2,
+            styles.textSupporting,
+            theme.icon,
+            translate,
+        ],
     );
 
-    const toggleCategory = (category: PolicyForList) => {
-        setSelectedCategories((prev) => ({
-            ...prev,
-            [category.value]: !prev[category.value],
-        }));
+    const toggleCategory = (category: PolicyOption) => {
+        setSelectedCategories((prev) => {
+            if (prev[category.keyForList]) {
+                const {[category.keyForList]: omittedCategory, ...newCategories} = prev;
```

Comment:
What is `omittedCategory` and are we reusing it anywhere else?

Reply:
I suggest using a destructuring pattern to exclude specific categories we don't want to include in the object. This approach allows us to keep only the selected categories in the list, rather than maintaining unselected categories with a value of false. WTYT?

Reply:
What about something like this? https://github.com/Expensify/App/pull/37717/files#diff-bc07e9853f8d016b7882bd8bfc8c58153e0cee5b3b5f5f3ba0365ac20b59f8deR129-R135

Reply:
@luacmartins I'm happy to bring back storing categories in the array. But you mentioned in first PR that we might have thousands of categories, iterating through objects probably might take in this case dozen milliseconds in low-end devices

Reply:
Hmm that's a good point. I think we can leave the logic as is then.